### ISSUE TITLE
Allow `*testing.T` before `context.Context` in the context parameter check

### ISSUE THREAD

**Issue by euank** [mgechev/revive#605]
This issue mirrors the upstream one [here](https://github.com/golang/lint/issues/422), which ended up stalling out due to upstream abandoning golint.

The specific case I'm referring to looks like so:

```go
func TestFoo(t *testing.T) {
  ctx, cancel := context.WithCancel(context.Backgrdoun())
  defer cancel()
  t.Run("subtest1", func(t *tesitng.T) {
    ctx, cancel := context.WithCancel(ctx)
    fooSubtest1(t, ctx)
  })
}

func fooSubtest1(t *testing.T, ctx context.Context) {
  // do things that require a context
}
```

Currently, the `context-as-argument` lint will complain that `fooSubtest1` should be written with the signature `func(context.Context, *testing.T)` instead.

To me, it's idiomatic for `*testing.T` to remain the first argument of `fooSubtest` since it's effectively being curried in. Ideally, if this language supported currying, we'd do something more akin to `subtest1 := t.Run("subtest1", fooSubtest1); subtest1(ctx)`, but without language support for currying, it seems like manually passing through the `t` first makes more sense.

So, that leads me to the question for this issue:
Should we update revive to allow this? Is there some other hoops we should jump through first?

I'm happy to provide the code changes necessary to make the current context lint accept this (and have them written up [over here](https://github.com/euank/revive/commit/9a452fbac3ed67c59ca37fe73404b99729d192ba) in fact.)
